packages/service-library/src/servicelib/logging_utils.py

```diff
@@ -52,25 +52,32 @@ class CustomFormatter(logging.Formatter):
     2. Overrides 'filename' with the value of 'file_name_override', if it exists.
     """
 
+    def __init__(self, fmt, color_log_enabled=False):
+        super().__init__(fmt)
+        self.color_log_enabled: bool = color_log_enabled
+
     def format(self, record):
         if hasattr(record, "func_name_override"):
             record.funcName = record.func_name_override
         if hasattr(record, "file_name_override"):
             record.filename = record.file_name_override
 
-        # add color
-        levelname = record.levelname
-        if levelname in COLORS:
-            levelname_color = COLORS[levelname] + levelname + NORMAL
-            record.levelname = levelname_color
-        return super().format(record)
+        if self.color_log_enabled == True:
+            levelname = record.levelname
+            if levelname in COLORS:
+                levelname_color = COLORS[levelname] + levelname + NORMAL
+                record.levelname = levelname_color
+        return super().format(record).replace("\n", "\\n")
 
 
 # SEE https://docs.python.org/3/library/logging.html#logrecord-attributes
-DEFAULT_FORMATTING = "%(levelname)s: [%(asctime)s/%(processName)s] [%(name)s:%(funcName)s(%(lineno)d)]  -  %(message)s"
+DEFAULT_FORMATTING = "log_level=%(levelname)s | log_timestamp=%(asctime)s | log_source=%(name)s:%(funcName)s(%(lineno)d) | log_msg=%(message)s"
```

Comment:
ok that makes it very verbose. but I guess that is the way for getting these in graylog.